### PR TITLE
Feat/register+email

### PR DESCRIPTION
**Pull request by DaniilVolobuiev**
## Describe your changes

> feat: Registration. Now user data can be saved in database.
> feat: User account email verification. User will receive an email and after following the link in the email user account wil be verified.
> feat: Google oAuth. User can login via Google and set up his info later in the next form.

## Issue ticket code (and/or) and link

> [Link to JIRA ticket](https://smartdc23.atlassian.net/browse/SMAR-30) 30
> [Link to JIRA ticket](https://smartdc23.atlassian.net/browse/SMAR-25) 25
> [Link to JIRA ticket](https://smartdc23.atlassian.net/browse/SMAR-35) 35
### General

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Functions is public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

@DaniilVolobuiev, @lyudad

### CODE REVIEW

**Comment by SwiftXLight** [comment]
auth.service registration don't have try catch
we should use Query Builder in services
why you delete .env.example?
why so much commented code?
please ucomment doctor controllers and services
if you decide to create future feature for login and validation and insert it as commented code please provide description for this

File: src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -1,13 +1,71 @@
...
+    return this.authService.updateGoogleDoctorHandler(
+      updateGoogleDoctorDto,
+      req.headers.authorization,
+    );
+  }
+
+  @ApiOperation({ summary: 'Account activation' })
+  @ApiResponse({ status: 201, type: Doctor })
+  @Get('/activation/:link')
+  @Redirect('https://nestjs.com') // mock value
```
Comment (lyudad) [inline src/modules/auth/auth.controller.ts]: if you will change it in the feature left //TODO you comment here
so in the future you can search by TODO word

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -35,6 +49,44 @@ export default class AuthService {
     });
   }
 
+  private readonly accessTokenCookieOptions: CookieOptions = {
+    maxAge: this.configService.get('ACCESS_TOKEN_MAX_AGE'),
+    httpOnly: true,
+    domain: 'localhost',
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: move to env variable

File: src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -49,14 +101,174 @@ export default class AuthService {
     }
   }
 
-  async login(doctorDto: LoginDoctorDto): Promise<{ token: string }> {
-    const doctor = await this.validateUser(doctorDto);
-    return this.generateToken(doctor);
+  private async sendActivationMail(to: string, link: string): Promise<void> {
+    try {
+      await this.transporter.sendMail({
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: make separate service for emails and templates, to have one place where you can change it if needed

File: src/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -49,14 +101,174 @@ export default class AuthService {
...
+
+    const doctor = await this.validateDoctorFromGoogle(googleDoctor);
+
+    const accessToken = this.jwtService.sign(
+      { ...doctor },
+      { expiresIn: '8h' },
+    );
+
+    res.cookie('accessToken', accessToken, this.accessTokenCookieOptions);
+    res.redirect('http://localhost:4200/');
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: move to env

File: src/modules/auth/auth.controller.ts [hunk 5]
```diff
@@ -1,13 +1,71 @@
...
+    return this.authService.updateGoogleDoctorHandler(
+      updateGoogleDoctorDto,
+      req.headers.authorization,
+    );
+  }
+
+  @ApiOperation({ summary: 'Account activation' })
+  @ApiResponse({ status: 201, type: Doctor })
+  @Get('/activation/:link')
+  @Redirect('https://nestjs.com') // mock value TODO
```
Comment (lyudad) [inline src/modules/auth/auth.controller.ts]: //TODO change value

**Pull request merged**